src/components/screen-header/use-ticket-info.tsx

```diff
@@ -0,0 +1,53 @@
+import {useAnalytics} from '@atb/analytics';
+import {
+    PreassignedFareProduct,
+  findReferenceDataById,
+  isOfFareProductRef,
+  useFirestoreConfiguration,
+} from '@atb/configuration';
+import {RootNavigationProps} from '@atb/stacks-hierarchy';
+import {FareContract, useTicketingState} from '@atb/ticketing';
+import {useNavigation} from '@react-navigation/native';
+
+type TicketInfo = {
+  navigateToTicketInfoScreen: () => void;
+  fareContract: FareContract | undefined;
+  preassignedFareProduct: PreassignedFareProduct | undefined;
+};
+
+export const useTicketInfo = (orderId: string): TicketInfo => {
+  const {findFareContractByOrderId} = useTicketingState();
+  const fareContract = findFareContractByOrderId(orderId);
+  const firstTravelRight = fareContract?.travelRights[0];
+
+  const {preassignedFareProducts} = useFirestoreConfiguration();
+  const preassignedFareProduct = findReferenceDataById(
+    preassignedFareProducts,
+    isOfFareProductRef(firstTravelRight) ? firstTravelRight.fareProductRef : '',
+  );
+
+  const analytics = useAnalytics();
+  const navigation = useNavigation<RootNavigationProps>();
```

Comment:
We should not use `useNavigation` like this. We have typing in the navigation-types files for the navigation objects, to specify the navigations allowed, but by using `useNavigation` with generics this typing is overridden. 

So it is mainly screens that should use the navigation object, other things like components and hooks should be decoupled from the navigation object. (With exception for simpler stuff like `navigation.goBack()` which isn't related to the typing of the navigation object.)

It might seem like a little overkill with such a rule that we shouldn't use `useNavigation` where we want the navigation object, but by experience the navigation around in the app was a little out of control and we lost type safety before we introduced this rule.

The [development guidelines](https://github.com/AtB-AS/docs-private/blob/main/kundevendt-development-guidelines.md) mentions this when describing screen wrappers, but I see that it should probably be clearer that the same rule applies to hooks and components 😊

Reply:
resolved in https://github.com/AtB-AS/mittatb-app/pull/4251